### PR TITLE
Gawr 667 oslo endpoint adres

### PR DESCRIPTION
**Pull request by YusufMavzer**
- [x] Address V2
- [x] Address V2 is oslo compliant
- [x] Address V2 has gml property
- [x] Address V2 moved 2 properties into adrespositie <- (positieGeometrieMethode, positieSpecificatie)
- [x] Projection Handling
**AddressMatch V2  (out-of-scope)**
**AddressMatch V2 is oslo compliant (out-of-scope)**

### CODE REVIEW

File: src/AddressRegistry.Api.Oslo/Address/AddressController.cs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            [FromServices] LegacyContext context,
+            [FromServices] SyndicationContext syndicationContext,
+            [FromServices] IOptions<ResponseOptions> responseOptions,
+            [FromRoute] int persistentLocalId,
+            [FromRoute] Taal? taal,
+            CancellationToken cancellationToken = default)
+        {
+            // TODO: Check if an etag is provided, if it is: check if the projections have caught up
+            //       and temporarily for crab, check if the crab position has caught up as well
+            // -> could possibly also be done in an filter attribute
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/Address/AddressController.cs:68]: can be removed

File: src/AddressRegistry.Api.Oslo/AddressMatch/AddressMatchContext.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+namespace AddressRegistry.Api.Oslo.AddressMatch
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/AddressMatch/AddressMatchContext.cs:1]: Don't think addressmatch is needed atm

File: src/AddressRegistry.Api.Oslo/AddressMatch/AddressMatchController.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
+namespace AddressRegistry.Api.Oslo.AddressMatch
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/AddressMatch/AddressMatchController.cs:1]: Same addressmatch comment

File: src/AddressRegistry.Api.Oslo/Infrastructure/Options/ResponseOptions.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        public string PostInfoNaamruimte { get; set; }
+        public string VolgendeUrl { get; set; }
+        public string DetailUrl { get; set; }
+        public string PostInfoDetailUrl { get; set; }
+        public string StraatnaamDetailUrl { get; set; }
+        public string GemeenteDetailUrl { get; set; }
+        public string GebouweenheidDetailUrl { get; set; }
+        public string PerceelDetailUrl { get; set; }
+        public double SimilarityThreshold { get; set; }
+        public int MaxStreetNamesThreshold { get; set; }
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/Infrastructure/Options/ResponseOptions.cs:17]: Addressmatch related

File: src/AddressRegistry.Api.Oslo/Infrastructure/StringReplaceExtension.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
+namespace AddressRegistry.Api.Oslo.Infrastructure
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/Infrastructure/StringReplaceExtension.cs:1]: Also used for addressmatch iirc

File: src/AddressRegistry.Api.Oslo/appsettings.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
+{
+  "ConnectionStrings": {
+    "Events": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;",
+    "LegacyProjections": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;",
+    "SyndicationProjections": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;",
+    "BuildingLegacyProjections": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;"
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/appsettings.json:6]: `Events` and `Building not needed

File: src/AddressRegistry.Api.Oslo/appsettings.json [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    "Enabled": false,
+    "Debug": false,
+    "ServiceName": "AddressRegistry Oslo"
+  },
+
+  "BaseUrl": "https://api.staging-basisregisters.vlaanderen/",
+
+  "Configuration": {
+    "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;"
+  },
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/appsettings.json:19]: Can't remember what this is used for? Any idea?

File: src/AddressRegistry.Api.Oslo/appsettings.json [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  "StraatNaamNaamruimte": "https://data.vlaanderen.be/id/straatnaam",
+  "DetailUrl": "https://basisregisters.vlaanderen.be/api/v2/adressen/{0}",
+  "VolgendeUrl": "https://basisregisters.vlaanderen.be/api/v2/adressen?offset={0}&limit={1}",
+  "PostInfoDetailUrl": "https://basisregisters.vlaanderen.be/api/v2/postinfo/{0}",
+  "GemeenteDetailUrl": "https://basisregisters.vlaanderen.be/api/v2/gemeenten/{0}",
+  "StraatnaamDetailUrl": "https://basisregisters.vlaanderen.be/api/v2/straatnamen/{0}",
+  "GebouweenheidDetailUrl": "https://basisregisters.vlaanderen.be/api/v2/gebouweenheden/{0}",
+  "PerceelDetailUrl": "https://basisregisters.vlaanderen.be/api/v2/percelen/{0}",
+  "SimilarityThreshold": 75.0,
+  "MaxStreetNamesThreshold": 100,
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/appsettings.json:37]: Addressmatch config

File: src/AddressRegistry.Api.Oslo/appsettings.json [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+  "BaseUrl": "https://api.staging-basisregisters.vlaanderen/",
+
+  "Configuration": {
+    "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;"
+  },
+
+  "Idempotency": {
+    "ConnectionString": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.AddressRegistry;Trusted_Connection=True;"
+  },
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/appsettings.json:23]: Can be removed

File: src/AddressRegistry.Api.Oslo/paket.references [hunk 10]
```diff
@@ -0,0 +1,13 @@
+AspNetCore.HealthChecks.SqlServer
+
+Be.Vlaanderen.Basisregisters.Api
+Be.Vlaanderen.Basisregisters.EventHandling.Autofac
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/paket.references:4]: Can you check if this package is still needed?

File: src/AddressRegistry.Api.Oslo/paket.references [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+Be.Vlaanderen.Basisregisters.DataDog.Tracing.Autofac
+Be.Vlaanderen.Basisregisters.Shaperon
+
+Microsoft.SyndicationFeed.ReaderWriter
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/paket.references:8]: This won't be needed

File: src/AddressRegistry.Api.Oslo/paket.references [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+
+AsyncEnumerator
```
Comment (ArneD) [inline src/AddressRegistry.Api.Oslo/paket.references:10]: Don't think this is needed either

File: src/AddressRegistry.Api.Oslo/appsettings.json (hunk 7 above)
Comment (YusufMavzer) [reply src/AddressRegistry.Api.Oslo/appsettings.json:19]: nope it's a copy paste from legacy

File: src/AddressRegistry.Api.Oslo/AddressMatch/AddressMatchContext.cs (hunk 2 above)
Comment (YusufMavzer) [reply src/AddressRegistry.Api.Oslo/AddressMatch/AddressMatchContext.cs:1]: They wanted AddressMatch to be V2 as well but making it Oslo compliant is still in refinement stage

File: src/AddressRegistry.Api.Oslo/paket.references (hunk 10 above)
Comment (YusufMavzer) [reply src/AddressRegistry.Api.Oslo/paket.references:4]: not needed

**Pull request merged**